docs/developer/architecture.rst

```diff
@@ -0,0 +1,260 @@
+System Architecture
```

Comment:
Do you want to delete the old version of the Architecture docs? Were there any changes from that?